### PR TITLE
markers: recognize empty `python_version` markers

### PR DESCRIPTION
**Pull request by radoering**
Without this fix, the intersection of `python_version > "3.8"` and `python_version < "3.9"` is not detected as empty marker.

Related to: python-poetry/poetry#10249

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

## Summary by Sourcery

Tests:
- Add tests to verify that the intersection of mutually exclusive markers is correctly identified as empty.

### CODE REVIEW

File: tests/version/test_markers.py [hunk 1]
```diff
@@ -171,6 +171,22 @@ def test_single_marker_intersect() -> None:
...
+@pytest.mark.parametrize(
+    ("marker1", "marker2"),
+    [
+        ('sys_platform == "win32"', 'sys_platform == "linux"'),
+        ('sys_platform != "win32"', 'sys_platform == "win32"'),
+        ('python_version >= "3.6"', 'python_version < "3.6"'),
+        ('python_version == "3.6"', 'python_version == "3.7"'),
+        ('python_version > "3.6"', 'python_version <= "3.6"'),
+        ('python_version >= "3.6"', 'python_version < "3.6"'),
+        ('python_version > "3.6"', 'python_version < "3.7"'),
```
Comment (radoering) [inline tests/version/test_markers.py:183]: The last test is the relevant one but since I did not find such a basic test function that tests for empty single marker intersections, I just added some additional tests.

File: tests/version/test_markers.py [hunk 2]
```diff
@@ -2260,16 +2271,16 @@ def test_intersection_no_endless_recursion() -> None:
         ' and extra != "postgis"'
     )
     m2 = parse_marker(
-        'python_version > "3.12" and python_version < "3.13" or extra != "databricks"'
+        'python_version >= "3.12" and python_version < "3.13" or extra != "databricks"'
```
Comment (radoering) [inline tests/version/test_markers.py:2274]: This marker is actually empty and does not trigger a potential endless recursion anymore. I checked that the changed marker triggers the recursion error without the fix from #832.

**Review by Secrus — APPROVED** [review]
